File: src/dotnet/Ansight.Tools.VisualTree/ShowOverlayTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class ShowOverlayTool : ITool
{
    public string Category => "ui";

    public ToolPolicy Policy => ToolPolicy.Write;

    public string Id => VisualTreeToolIds.ShowOverlay;

    public string Name => "Show Overlay";

    public string Description => "Draws an input-transparent diagnostic overlay over the active app window.";

    public string Keywords => "ui overlay highlight box rectangle diagnostic";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.ShowOverlayArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.OverlayResult;

    public Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        return VisualTreeSupport.ShowOverlayAsync(arguments);
    }
}

File: src/ios/Sources/AnsightToolsVisualTree/ShowOverlayTool.swift
import AnsightCore
import Foundation

public final class ShowOverlayTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.showOverlay,
            name: "Show Overlay",
            description: "Draws an input-transparent diagnostic overlay over the active app window.",
            category: "ui",
            policy: .write,
            keywords: "ui overlay highlight box rectangle diagnostic",
            argumentsSchema: AnsightVisualTreeToolSchemas.showOverlayArguments,
            resultSchema: AnsightVisualTreeToolSchemas.overlayResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeOverlaySupport.showOverlay(arguments: arguments)
    }
}
